### PR TITLE
Update package.json to exclude source code from NPM package

### LINKED ISSUE #200: Source code should not be packed into the NPM package
**Is your suggestion related to an experience ? Please describe.**

The bundled package in the NPM registry contains the uncompiled src directory as well in it, accounting for about 45KB/700KB, even though it brings no value to the end users.

image

In addition, the compiled types need to be in the dist directory itself or in a meaningful folder structure (eg: `dist/types`)

Screenshot 2023-08-07 at 21 44 25

**Describe the improvement**

src/ folder shouldn't be packed into the NPM package.

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by s-vamshi**
Resolves #200 

src folder shouldn't be added in package and compiled types should be inside dist/types 

Added .npmignore file to ignore src and added decelerationDir key in tsconfig.json to put complied types in dist/types/src

### CODE REVIEW

File: lib/.npmignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+src/
+!/dist/types/src
```
Comment (pavinduLakshan) [inline lib/.npmignore:2]:
```suggestion
!/dist/types/src

```

Do we need ! here?

File: lib/tsconfig.json [hunk 2]
```diff
@@ -23,4 +24,4 @@
     },
     "compileOnSave": false,
     "exclude": ["node_modules", "test-configs", "src/**/tests/*", "**/*.test.ts", "**/node_modules"]
-}
+}
```
Comment (pavinduLakshan) [inline lib/tsconfig.json:27]: Let's add newline.

```suggestion
}

```

**Review by pavinduLakshan — COMMENTED** [review]
Let's also exclude tsbundleinfo as well from being packed into the SDK bundle.

File: lib/.npmignore (hunk 1 above)
Comment (s-vamshi) [reply lib/.npmignore:2]: Yes as /src would exclude all directory with the src name, so we need to exclude src inside types

File: lib/.npmignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+dist/tsconfig.tsbuildinfo
```
Comment (NipuniBhagya) [inline lib/.npmignore:3]: Let's add a new line here as well.

**Comment by s-vamshi** [comment]
Hey @NipuniBhagya @pavinduLakshan done with the changes

File: lib/.npmignore (hunk 3 above)
Comment (NipuniBhagya) [inline lib/.npmignore:3]: Better to exclude the development related files as well like,

tsconfig.json
webpack.config.js
babel.config.js
.eslintrc.js
.eslintignore

Comment (s-vamshi) [reply lib/.npmignore:3]: Done with the changes

**Comment by s-vamshi** [comment]
@NipuniBhagya can you please review this now made the changes

**Comment by brionmario** [comment]
Hi @s-vamshi,

While `.npmignore` seems like a viable option to overcome the issue, i think we can resolve this with a simple `files` entry in the `package.json`.

Have you considered that option?

IMO, since we don't go into granular level to which files in include, rather we can simply include `dist`, `package.json`, `README`, better to go for the `files` option.

Additionally, we could also copy the `LICENSE` file at the root to the `lib` folder, and include that as well in the distribution.

WDYT?

**Comment by s-vamshi** [comment]
Hey @brionmario, I agree with you, as lib folder didn't have .gitignore I thought of going with .npmignore. 

I will add files section in package.json and push them!

**Comment by brionmario** [comment]
> Hey @brionmario, I agree with you, as lib folder didn't have .gitignore I thought of going with .npmignore.
> 
> I will add files section in package.json and push them!

Cool 🙌

Lets duplicate the `LICENSE` inside `lib` and include that as well :)

**Comment by s-vamshi** [comment]
Sure 🚀

File: lib/package.json [hunk 4]
```diff
@@ -72,4 +79,5 @@
         "react": ">=16.8",
         "react-dom": ">=16.8"
     }
+
```
Comment (pavinduLakshan) [inline lib/package.json:82]:
```suggestion
```

Comment (s-vamshi) [reply lib/package.json]: done with the changes @pavinduLakshan

Comment (pavinduLakshan) [reply lib/package.json]: Thanks @s-vamshi , please update the PR title as well to accurately describe the changes.

Comment (s-vamshi) [reply lib/package.json]: hey @pavinduLakshan updated pr title

**Comment by s-vamshi** [comment]
Hey @brionmario @NipuniBhagya @pavinduLakshan I have made the changes as suggested can you please review this PR?

**Comment by s-vamshi** [comment]
Hey @brionmario @NipuniBhagya @pavinduLakshan can you please review this PR?

**Review by pavinduLakshan — APPROVED** [review]
LGTM

**Comment by brionmario** [comment]
LGTM too.

Merged! Thanks @s-vamshi for the contribution 🎉

**Comment by brionmario** [comment]
🚢 Shipped with [v5.2.0](https://github.com/asgardeo/asgardeo-auth-react-sdk/releases/tag/v5.2.0)

**Comment by pavinduLakshan** [comment]
Thanks @s-vamshi for the contribution. This just reduced the package size by 10% (744KB -> 659KB). Great work!

**Comment by s-vamshi** [comment]
> Thanks @s-vamshi for the contribution 🎉
> 
> Looks like the fix is working as expected. If time permits, please try the released version in a React app and do a regression test round!
> 
> Screenshot 2024-10-14 at 19 15 26
> 

Sure I will do regression testing

**Comment by NipuniBhagya** [comment]
Hi @s-vamshi,

🎉 Thank you for your contribution! 🎉

We appreciate your effort and the time you've taken to submit this PR. As part of the Hacktoberfest process, please make sure to fill out the form [1] shared to ensure we can properly track your contribution.

[1] - https://docs.google.com/forms/d/1RT5fdrj7DvJCtzHOpvAIbN-4d08RswCOK_GZswZKR8w/viewform?edit_requested=true

Once again, thank you for your support, and we're excited to have you involved in our project! 🙌

**Pull request merged**